### ISSUE TITLE
[clangd] False positive ODR violations with C++ modules

### ISSUE THREAD

**Issue by rdong8** [llvm/llvm-project#196678]
This is reproducible with clangd 19-22.

`print.cppm`:

```cpp
module;

#include <utility>

#include <llvm/ADT/StringRef.h>
#include <llvm/Support/FormatVariadic.h>
#include <llvm/Support/raw_ostream.h>

export module print;

template <typename... Args> auto println(llvm::raw_ostream &stream, llvm::StringRef fmt, Args &&...args) -> void
{
    llvm::formatv(fmt.data(), std::forward<Args>(args)...).format(stream);
    stream << '\n';
}
```

`uses_print.cppm`:

```cpp
module;

// This header must be included to surface the error
#include <llvm/Support/FormatProviders.h>

export module uses_print;

// Error here:
// In included file: 'llvm::raw_ostream' has different definitions in different modules; first difference is definition in module 'print.<global>' found constructor with body
// clang(module_odr_violation_record)
import print;

auto foo(llvm::raw_ostream &) -> void
{
}
```

Full reprod [here](https://github.com/rdong8/cpp_project/tree/modules-debug).

**Comment by ArcsinX** [comment]
Can you please try the latest clangd? You can find one here https://github.com/clangd/clangd/releases (take the latest unstable snapshot)

**Comment by rdong8** [comment]
Thanks, seems to have been fixed
